sdks/python/apache_beam/internal/apiclient_test.py

```diff
@@ -35,8 +34,8 @@ def test_create_application_client(self):
 
   def test_default_job_name(self):
     job_name = apiclient.Job.default_job_name(None)
-    regexp = 'beamapp-[a-z]*-[0-9]{10}-[0-9]{6}'
-    self.assertTrue(re.match(regexp, job_name))
+    regexp = 'beamapp-.*-[0-9]{10}-[0-9]{6}'
```

Comment:
That is a very good question.

That group matches the user name, it can contain arbitrary things and not all of them will make legal job names. If it is an invalid name it will fail in the validation phase but I think the default name should never be invalid.

I think this should be fixed in the default name creation. I do not think it needs to be fixed in this PR, but we at least need a JIRA for it.